src/ui/blockUi-editor.js

```diff
@@ -114,6 +118,145 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
+                this: "{that}.dom.uploadButton",
+                method: "click",
+                args: ["{that}.events.onMediaUploadRequested.fire"],
+                namespace: "bindOnMediaUploadRequested"
+            }
+        },
+        components: {
+            // handles uploading a single file for storage
+            singleFileUploader: {
+                type: "sjrk.storyTelling.block.singleFileUploader",
+                createOnEvent: "{templateManager}.events.onTemplateRendered",
+                container: "{editor}.dom.singleFileUploader",
+                options: {
+                    modelRelay: {
+                        "uploadState": {
+                            target: "{editor}.model",
+                            singleTransform: {
+                                type: "fluid.transforms.valueMapper",
+                                defaultInputPath: "uploadState",
+                                match: {
+                                    "ready": {
+                                        outputValue: {
+                                            previewVisible: true,
+                                            progressAreaVisible: false,
+                                            responseAreaVisible: false,
+                                            uploadButtonDisabled: false
+                                        }
+                                    },
+                                    "uploading": {
+                                        outputValue: {
+                                            previewVisible: false,
+                                            progressAreaVisible: true,
+                                            responseAreaVisible: false,
+                                            uploadButtonDisabled: true
+                                        }
+                                    },
+                                    "errorReceived": {
+                                        outputValue: {
+                                            previewVisible: true,
+                                            progressAreaVisible: false,
+                                            responseAreaVisible: true,
+                                            uploadButtonDisabled: false
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    },
+                    selectors: {
+                        fileInput: "{that}.container"
+                    },
+                    model: {
+                        fileObjectUrl: "{block}.model.mediaUrl",
+                        storyId: "{editor}.storyId"
+                    },
+                    listeners: {
+                        "{editor}.events.onMediaUploadRequested": {
+                            func: "{that}.events.onFileSelectionRequested.fire",
```

Comment:
I don't think you need to add the `fire` segment at the end in the listener block.